online-shop/src/app/product-details/product-details.component.ts

```diff
@@ -1,22 +1,34 @@
-import {Component, OnInit} from '@angular/core';
-import {ActivatedRoute} from "@angular/router";
-import {Product} from "../../product";
-import {products} from "../../product";
-
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+import { Product } from '../../product';
+import {ProductService} from "../product.service";
 
 @Component({
   selector: 'app-product-details',
   templateUrl: './product-details.component.html',
-  styleUrls: ['./product-details.component.scss']
+  styleUrls: ['./product-details.component.scss'],
 })
 export class ProductDetailsComponent implements OnInit {
   id = 0;
-  product!: Product
-constructor(
-  private route: ActivatedRoute,
-) {}
+  product!: Product;
+  constructor(private route: ActivatedRoute, private productService: ProductService) {}
+
+
+
   ngOnInit(): void {
-   this.id = parseInt(this.route.snapshot.paramMap.get('id')!);
-   this.product = products.find(product => product.id === this.id)!;
+    this.id = parseInt(this.route.snapshot.paramMap.get('id')!);
+    this.productService.getProductById(this.id).subscribe(data => {
+      this.product = data;
+    })
   }
+
+deleteProduct() {
+    this.productService.deletProductById(this.id).subscribe();
```

Comment:
as we know miss clicks happens, and here the user won't be able to cancel his unwanted action.
When you will add material in the app, have a look here -> https://material.angular.io/components/dialog/overview
until then you can have a look here -> https://www.w3schools.com/jsref/met_win_confirm.asp